src/stores/movies.ts

```diff
@@ -0,0 +1,36 @@
+import { defineStore } from "pinia";
+import { getMovieList } from "@/api/movieService";
+import type { Movie, Genre } from "@/types/movie";
+
+interface RootState {
+  movieList: Movie[];
```

Comment:
since movieList can be null you could specify it here and avoid using casting with `as`

```suggestion
  movieList: Movie[] | null;
```

Reply:
I have changed this to `[]`. One of depending getters whast flagging in TS for possiblly null when `null` was applied

Reply:
You have two options here that I think solve the problem in a way that will let you levarage typescript typing better.

Option 1: we assume movieList can take `null` value and in the getter we take care of the nullish values and return empty array for example
Option 2: we make assumption movieList should always be an array, but then we can't initialize it with null default value, we can use empty array instead `[]`.

That way you specify the type in advance and then stick to the type you designed :)

Reply:
Thank you for nice explanation. I always This is the part that I found always a bit confusing. The way you explained that way we initialise value will dictate how we validate for said values in rest of app really helped :)